### PR TITLE
feat(optimizer)!: Annotate and add tests for snowflake REPLACE and SPACE functions

### PR DESCRIPTION
**Pull request by fivetran-amrutabhimsenayachit**
Annotate and add tests for snowflake REPLACE and SPACE functions.

Documentation:
https://docs.snowflake.com/en/sql-reference/functions/replace
https://docs.snowflake.com/en/sql-reference/functions/space

### CODE REVIEW

File: sqlglot/dialects/dialect.py [hunk 1]
```diff
@@ -775,6 +775,8 @@ class Dialect(metaclass=_Dialect):
             exp.GroupConcat,
             exp.Initcap,
             exp.Lower,
+            exp.Replace,
+            exp.Space,
```
Comment (geooo109) [inline sqlglot/dialects/dialect.py]: We should move these into the Snowflake dialect.
Adding these here means that we have checked the return type for most of the dialects. (which is harder to do, and currently we want to avoid it)

File: tests/fixtures/optimizer/annotate_functions.sql [hunk 2]
```diff
@@ -1747,6 +1747,22 @@ VARCHAR;
...
+REPLACE('hello', 'old', NULL);
+VARCHAR;
+
+# dialect: snowflake
+SPACE(5);
+VARCHAR;
+
+# dialect: snowflake
+SPACE(tbl.int_col);
+VARCHAR;
```
Comment (geooo109) [inline tests/fixtures/optimizer/annotate_functions.sql:1764]: Can you also add a test for `SPACE(NULL)` ?
